.github/workflows/test-aks.yaml

```diff
@@ -0,0 +1,362 @@
+          echo "TEST_EKS_CLUSTER_NAME=eks-${RUN_IDENTIFIER}" >> $GITHUB_OUTPUT
+          echo "RUN_TEST=${RUN_TEST}" >> $GITHUB_OUTPUT
+          echo "ENABLE_DAPR=${ENABLE_DAPR}" >> $GITHUB_OUTPUT
+      - name: Checkout code
+        if: steps.gen-id.outputs.RUN_TEST == 'true'
+        uses: actions/checkout@v3
+      - name: Ensure inputs.version is valid semver
+        if: steps.gen-id.outputs.RUN_TEST == 'true' && inputs.version != ''
+        run: |
+          python ./.github/scripts/validate_semver.py ${{ inputs.version }}
+      - name: Setup Node
+        if: steps.gen-id.outputs.RUN_TEST == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: az CLI login
+        if: steps.gen-id.outputs.RUN_TEST == 'true'
+        run: |
+          az login --service-principal \
+            --username ${{ secrets.AZURE_SP_TESTS_APPID }} \
+            --password ${{ secrets.AZURE_SP_TESTS_PASSWORD }} \
+            --tenant ${{ secrets.AZURE_SP_TESTS_TENANTID }}
+      - name: Configure AWS
+        if: steps.gen-id.outputs.RUN_TEST == 'true' && matrix.credential == 'aws'
+        run: |
+          aws configure set aws_access_key_id ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws configure set aws_secret_access_key ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws configure set region ${{ env.AWS_REGION }}
+          aws configure set output json
+      - name: Get kubeconf credential for AKS cluster
+        if: steps.gen-id.outputs.RUN_TEST == 'true'
+        run: |
+          az aks get-credentials \
+            --subscription ${{ secrets.AZURE_SUBSCRIPTIONID_TESTS }} \
+            --resource-group ${{ env.AKS_RESOURCE_GROUP }} \
+            --name ${{ env.AKS_CLUSTER_NAME }}
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build images
+        if: steps.gen-id.outputs.RUN_TEST == 'true' && matrix.images != ''
+        run: |
+          # split images and directories into arrays
+          IFS=',' read -ra images <<< "${{ matrix.images }}"
+          IFS=',' read -ra directories <<< "${{ matrix.directories }}"
+          echo "Building images: $images"
+          echo "Directories: $directories"
+          for index in "${!images[@]}"; do
+            image=${images[$index]}
+            directory=${directories[$index]}
+            echo "Building image $image from directory $directory"
+            docker build -t ghcr.io/radius-project/$image:${{ github.run_id }} $directory
+            docker tag ghcr.io/radius-project/$image:${{ github.run_id }} ghcr.io/radius-project/$image:${{ github.run_id }}
+            # We need to think about how we can delete these images at the end of the run
+            docker push ghcr.io/radius-project/$image:${{ github.run_id }}
+          done
```

Comment:
This is where we build the necessary images and push them to GHCR. Would love to get ideas on this approach. I see that for some tests we are using the latest tag but for some we are building images. What would the best approach be here?